int32 enemy_dist
int32 enemy_yaw
int32 enemy_pitch
int32 mode 
